crypto/poly1305/poly1305_arm.c

```diff
@@ -179,25 +179,28 @@ static const alignas(16) fe1305x2 zero;
 struct poly1305_state_st {
   uint8_t data[sizeof(fe1305x2[5]) + 128];
   uint8_t buf[32];
-  unsigned int buf_used;
+  size_t buf_used;
   uint8_t key[16];
 };
 
+OPENSSL_STATIC_ASSERT(
+    sizeof(struct poly1305_state_st) + 63 <= sizeof(poly1305_state),
+    "poly1305_state isn't large enough to hold aligned poly1305_state_st.");
```

Comment:
Second argument is not a token. How is this passing?

Reply:
I don't plan to change this because no CI dimension can reach this code.

```c
#if !defined(__ARM_ARCH__)
# if defined(__CC_ARM)
#  define __ARM_ARCH__ __TARGET_ARCH_ARM
#  if defined(__BIG_ENDIAN)
#   define __ARMEB__
#  else
#   define __ARMEL__
#  endif
# elif defined(__GNUC__)

#elif defined(__ARMEL__) || defined(_M_ARM)
#define OPENSSL_32_BIT
#define OPENSSL_ARM

#if defined(OPENSSL_ARM) && !defined(OPENSSL_NO_ASM) && !defined(OPENSSL_APPLE)
#define OPENSSL_POLY1305_NEON
```

Reply:
We can't make decisions based on what can currently be reached and what cannot. If `OPENSSL_ARM` is enabled then the build fails, which it might in the future, who knows.
I also prefer the original assert. But as you know, it was not C99 compliant.

Reply:
Changed.

 It's not easy to enable `OPENSSL_ARM` because `OPENSSL_ARM` replies on `__ARMEL__ ` or `_M_ARM`, which depends on `__CC_ARM`. When CI adds this, the code change will be detected(validated).